### PR TITLE
Support the prank() family of cheatcodes

### PR DESCRIPTION
**Pull request by cd1m0**
What title says. This is a slightly more complex PR. To support pranks, we needed to keep them scoped to external call contexts. To do so we hotpatch the `Interpreter.run` function, and add callbacks that keep track of the external call stack. Then we add (first pending and then activated) pranks as additional properties on the EEI objects that we use to identify external call frames. Additionally we interpose on the CALLER and ORIGIN opcodes to implement the prank behavior

### CODE REVIEW

File: src/debug/foundry_cheatcodes.ts [hunk 1]
```diff
@@ -46,20 +102,161 @@ export const FAIL_MSG_DATA = Buffer.concat([
...
+
+    // Get any pending pranks for the current call frame
+    public getPendingPrank(): FoundryPrank | undefined {
+        if (this.eeiStack.length === 0) {
+            return undefined;
+        }
+
+        const eei = this.eeiStack[this.eeiStack.length - 1];
+
+        return (eei as any).pendingPrank;
```
Comment (blitz-1306) [inline src/debug/foundry_cheatcodes.ts:140]: Can be shortened to
```typescript
return (this.getEEI() as any).pendingPrank;
```
Not a blocker.

File: src/debug/foundry_cheatcodes.ts [hunk 2]
```diff
@@ -46,20 +102,161 @@ export const FAIL_MSG_DATA = Buffer.concat([
...
+
+    // Set the pending prank for the current call frame
+    public setPendingPrank(prank: FoundryPrank | undefined): void {
+        if (this.eeiStack.length === 0) {
+            return undefined;
+        }
+
+        const eei = this.eeiStack[this.eeiStack.length - 1];
+
+        (eei as any).pendingPrank = prank;
```
Comment (blitz-1306) [inline src/debug/foundry_cheatcodes.ts:151]: This also can be shortened with `this.getEEI()`.

**Review by blitz-1306 — APPROVED** [review]
Great!

File: src/debug/foundry_cheatcodes.ts (hunk 1 above)
Comment (cd1m0) [reply src/debug/foundry_cheatcodes.ts]: Fixed in [b29dbfe](https://github.com/ConsenSys/sol-dbg/pull/23/commits/b29dbfea1af90e8af6a85bc2fd2f4e67e6a8ca8a)

File: src/debug/foundry_cheatcodes.ts (hunk 2 above)
Comment (cd1m0) [reply src/debug/foundry_cheatcodes.ts]: Fixed in [b29dbfe](https://github.com/ConsenSys/sol-dbg/pull/23/commits/b29dbfea1af90e8af6a85bc2fd2f4e67e6a8ca8a)

**Pull request merged**